### ISSUE TITLE
Autosubmit is unable to run under BSC machines except bschubs.

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1482]
In GitLab by @jeronimoescribano on Dec 5, 2024, 16:16

Hello @dbeltrankyl and @kinow,

I have an issue posted in the slack hpcissues by @cmv212  that I guess it is better to follow here. Basically, I cannot run my experiments due to an error related to the database, following the error code.  Because it appears after this esarchive quouta issue, I'm not sure if it specific to experiments or something more general. 

### Autosubmit Version

3.15.19

### Expid affected(If applicable)

a82c a82d a82e a82f 

### Which task has issues? Where is the log(If applicable)

all experiment

### Summary

Since last weekend (esarchive quota issue) I have been unable of restart my experiments. 

### Steps to reproduce

autosumit run EXPID 

with EXPID=a82[cdef]

### What is the current bug behavior?

do not run.

### What is the expected correct behavior?

run experiment

### Relevant logs and/or screenshots(if applicable)

from the autosubmit01 and 02 machines: 

```
autosubmit_a82c.conf OK
Configuration files OK

 [CRITICAL] We have detected that there is another Autosubmit instance using the experiment
. Stop other Autosubmit instances that are using the experiment or delete autosubmit.lock file located on tmp folder [eCode=7000]

```

### Any other relevant information (if applicable)

- there is no autosubmit.lock file in the folder
- from the bsceshub it tries to run (at least takes more time to test), but it fails by memory once it runs, or before. For example: 
```
Job a82d_20180724_047_11_SIM is COMPLETED
Job a82d_20180724_047_11_SIM finished at 2024-12-05 15:15:12
[ERROR] Trace: [Errno 12] Cannot allocate memory
 [CRITICAL] There is a bug in the code, please contact via gitlab [eCode=7070]

```

or 
```
jescriba@ bscearth343:/esarchive/autosubmit> tail a82c/mywd/nohup.out
load impi/2021.10.0 (PATH, MANPATH, LD_LIBRARY_PATH)
load mkl/2023.2.0 (LD_LIBRARY_PATH)
load UCX/1.15.0 (PATH, LD_LIBRARY_PATH, LIBRARY_PATH, C_INCLUDE_PATH,
CPLUS_INCLUDE_PATH) 
load bsc/1.0 (PATH, MANPATH)
[eCode=6006]
[marenostrum5] Connection successful to host glogin1.bsc.es,glogin2.bsc.es,glogin3.bsc.es,glogin4.bsc.es
[ERROR] Trace:  
 [CRITICAL] This seems like a bug in the code, please contact AS developers [eCode=7070]
More info at https://autosubmit.readthedocs.io/en/v3.15.0/faq.html

```
I've tried without success:
- autosubmit recovery -f -s  (works) but not change the issue when I try to run
- autosubmit create
- autosubmit dbfix (  `Error: near line [all lines]: database is locked` )
- autosubmit pklfix
- manually temporal cleaning/removing/renaming of folders pkl and tmp 

As I said before, I cannot properly test this in the hub, because it seems that AS with this number of jobs fails by memory (sometimes killed by the OS). On contrary, the AS machines seems to allow experiments of about 11000 jobs (at maximum). If you think the experiment is too large, I cannot split more the workflow because it becomes unmanageable in practice (I'm running bi-monthly now, and I need 5 years... and repeat all runs 2 times more). In any case, the size limits of AS experiments could deserve a dedicated issue different to this one. 
 
fyi @cmv212 @gmontanepinto @albertvilabsc 

()

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 9, 2024, 08:38

Hello @jeronimoescribano 

Sorry the delay, I had a holiday on Thursday, 

If they ran before, it could be that the joblist is in a weird status, 

Can you try to remove the joblist pkl file from these experiments? `rm $expid/pkl/*.pkl` and rm `$expid/pkl/*.db`

After that, do an `autosubmit create` + `autosubmit recovery`

> On contrary, the AS machines seem to allow experiments of about 11000 jobs (at maximum)

If you're referring to the last AS seminar, this is using Autosubmit 4.1+. I don't know the maximum amount of jobs using AS 3.15.

**Comment by kinow** [comment]
In GitLab by @jeronimoescribano on Dec 9, 2024, 11:33

Hi @dbeltrankyl 

Thanks for taking a look the the issue!. I've tried that already and didn't worked. If I do it now: 

```
jescriba@ bscesautosubmit02:/esarchive/autosubmit/a82c/pkl> ls -l
total 0
jescriba@ bscesautosubmit02:/esarchive/autosubmit/a82c/pkl> autosubmit create a82c -np
Autosubmit is running with 3.15.19
 [CRITICAL] We have detected that there is another Autosubmit instance using the experiment
. Stop other Autosubmit instances that are using the experiment or delete autosubmit.lock file located on tmp folder [eCode=7000]
More info at https://autosubmit.readthedocs.io/en/v3.15.0/faq.html
jescriba@ bscesautosubmit02:/esarchive/autosubmit/a82c/pkl> rm ../tmp/autosubmit.lock
rm: cannot remove ‘../tmp/autosubmit.lock’: No such file or directory
jescriba@ bscesautosubmit02:/esarchive/autosubmit/a82c/pkl> 

```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 9, 2024, 11:51

Thanks @jeronimoescribano , 

(as commented through slack) Contacted @stamenminkov through #ces ( so more people is aware) channel slack, it may be a general issue with these machines

**Comment by kinow** [comment]
In GitLab by @stamenminkov on Dec 9, 2024, 12:49

mentioned in issue requests#2695

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 10, 2024, 10:55

Hello @stamenminkov , 

Any news on this? Or some estimated time ( for now, I'm just forwarding people who want to use Autosubmit to the hubs, but this may lead to an overload of the hubs )

Thanks!

**Comment by kinow** [comment]
In GitLab by @stamenminkov on Dec 10, 2024, 11:59

Good morning @dbeltrankyl,

Still checking this with Ops! Seems to be a pretty bizarre issue (and could be an `esarchive` thing)...

It's difficult to give some estimated time (we're pushing though), but I don't think it will be working for today either :(   

Will get back to you when we have more news!

Stamen

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 10, 2024, 12:02

Thanks for the news!

**Comment by kinow** [comment]
In GitLab by @V-Agudetse on Dec 11, 2024, 10:31

mentioned in issue requests#2462

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 11, 2024, 13:26

Updated the description, so other users can easily see this issue

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 12, 2024, 10:11

Hello @stamenminkov 

Any news on this? We plan to send a message if this take longer

**Comment by kinow** [comment]
In GitLab by @stamenminkov on Dec 12, 2024, 12:46

Hi @dbeltrankyl, still battling with Ops unfortunately... :confused:

**Comment by dbeltrankyl** [comment]
Hello all

The issue with the autosubmit01, autosubmit02 and workstations has been resolved. We recommend using these machines again whenever possible to avoid overloading the hubs.

We believe that the issue was caused by the non-working lock file service of the NFS( filesystem ).

Operation "restarted some services," and today, we found that the issue with Autosubmit on these machines was resolved.

Thanks to Francesco, Albert, Stamen and P.A for looking at it.

Apologies for the issues

Closing the issue
